api/src/test/java/uk/gov/dhsc/htbhf/claimant/utilities/DatabasePopulator.java

```diff
@@ -97,19 +124,53 @@ void generateClaimsAndPaymentCycles() {
             EligibilityStatus paymentEligibility = PAYMENT_ELIGIBILITY_STATUS.get(status);
             String ninoPrefix = NINO_PREFIX.get(status);
             Query query = entityManager.createNativeQuery("select count(*) from claimant where left(nino , 2) = '" + ninoPrefix + "'");
-            int ninoStartIndex = ((Number)query.getSingleResult()).intValue() + 1;
+            int ninoStartIndex = ((Number) query.getSingleResult()).intValue() + 10001; // ensure we have a non-zero number in the first 2 digits
             log.error("Creating {} claims in status {}, starting from {}", claims, status, ninoStartIndex);
+            List<Callable<Long>> jobs = new ArrayList<>(claims);
             for (int i = 0; i < claims; i++) {
-                Claim claim = createClaim(status, eligibilityStatus, ninoPrefix, ninoStartIndex + i);
-                if (paymentEligibility != null) {
-                    createPaymentCycle(cycleDuration, paymentEligibility, claim);
-                }
-                if (countCreated.incrementAndGet() % 100 == 0) {
-                    log.error("Created {} claims so far", countCreated.get());
-                }
-
+                Integer ninoNumber = ninoStartIndex + i;
+                Callable<Long> job = () -> {
+                    long start = System.currentTimeMillis();
+                    Claim claim = createClaim(status, eligibilityStatus, ninoPrefix, ninoNumber);
+                    if (paymentEligibility != null) {
+                        createPaymentCycle(cycleDuration, paymentEligibility, claim);
+                    }
+                    if (countCreated.incrementAndGet() % 1000 == 0) {
+                        log.error("Created {} claims so far", countCreated.get());
+                    }
+                    return System.currentTimeMillis() - start;
+                };
+                jobs.add(job);
             }
+            durations.addAll(invokeAllJobs(jobs));
         });
+
+        logDurations(durations);
+    }
+
+    private List<Long> invokeAllJobs(List<Callable<Long>> jobs) {
+        List<Long> durations = new ArrayList<>();
+        try {
+            List<Future<Long>> futures = fixedThreadPool.invokeAll(jobs);
+            for (Future<Long> future : futures) {
+                durations.add(future.get());
+            }
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+        return durations;
+    }
+
+    private void logDurations(List<Long> durations) {
```

Comment:
I found this quite difficult to understand, maybe it needs a refactor and some documentation to make it easier to understand?

Reply:
I added some comments, found it difficult to identify how I might refactor. Suggestions?

Reply:
Having spent some time with this, this is actually quite a good way to do this, it just takes a few minutes to understand what you're doing and why. I think if you just rename a couple of the variables then this becomes a bit clearer. My thoughts are:

- Rename `map` to `durationFrequencyMap`
- Rename `bucket` to `durationRoundedToNearest10` (I know this is lengthy so feel free to rename if you can think of something better)

I did then try to refactor out the generation of the logging String as I was initially confused by the `bucket + (bucketSize -1)` part, but that was quite tricky to make it more legible so I left that alone. 

One thing I might be tempted to do was to just put an example output as a comment to that method and just say: "This method outputs a frequency map of the durations and looks like this: " and then paste in one you've done before, I think that'll do it then.

Sorry, a bit of a long comment, but hopefully that covers everything.

Reply:
I should also say that I'm happy with this after that, so feel free to merge if I'm holding things up tomorrow. I would Approve but it'll only be dismissed when you do an update so no point.